packages/components/src/UISwitch.test.tsx

```diff
@@ -0,0 +1,34 @@
+import React from 'react';
+import { render } from '@testing-library/react';
+import userEvent from '@testing-library/user-event';
+import UISwitch from './UISwitch';
+
+function makeUISwitch(testId: string, onClick) {
+  return render(<UISwitch on onClick={onClick} data-testid={testId} />);
+}
+
+it('mounts and unmounts properly', () => {
+  function click() {
+    return true;
+  }
+
+  const testId = 'test id for UISwitch';
+  const testSwitch = makeUISwitch(testId, click);
+  testSwitch.unmount();
```

Comment:
With the above change, you don't need to create any parameters as you'll just use the defaults.
We also don't need to explicitly call `unmount()`, since that happens automatically during the `cleanup` phase with `testing-library/react`: https://testing-library.com/docs/react-testing-library/api/#cleanup

So then this test simply becomes:
```suggestion
  makeUISwitch();
```